### DISCUSSION TITLE
What are the ideal conditions to capture a region?

### DISCUSSION THREAD

**Discussion by Metazolid** [Nain57/Smart-AutoClicker#46] [Q&A] [answered]
More specifically; Is it better to select a small section of the screen or a large one? Is it better to have lots of detail in it or as few as possible? Does the amount of contrast within the region play a role in its accuracy of being detected?
And out of couriosity; When changing the tolerances, what kind of parameters are affectd by it?

**Comment by Nain57** [comment, marked as answer]
Hi !

I think that's a good question, as I can regularly see people in the issues tracker confused about how the application works. Here is a few points that can helps understand how Smart AutoClicker works. 


#### Scenario detection process

The application will iterate through the scenario events (clicks before v1.2.0) in the same order as the one configured by the user. 
For each of those events, it checks if the conditions are matched (given the conditions and the operator):
* If yes, the actions (click/swipe/pause) are applied and the events iteration is stopped.
* If no, do nothing and continue iterating.
Once the iteration is complete (or stopped) we wait for the next screen frame.


#### Condition matching:

The application compare each pixels colors of the configured image condition to the pixel colors of the current screen frame, at the exact same position. With a tolerance of 0, each pixels of the condition image must be the exact same as the pixels of the screen in order to match the condition. The tolerance is the amount (in %) of tolerated difference between the condition image pixels and the screen frame pixels.


#### How to select my conditions

* _Is it better to select a small section of the screen or a large one ?_ 
The detection will have the same accuracy with all images size, but bigger one will slow down the detection process, as there is more pixels to compare.

* _Is it better to have lots of detail in it or as few as possible ?_  _Does the amount of contrast within the region play a role in its accuracy of being detected ?_
The current algorithm doesn't care, only the pixels colors are important. 

* _When changing the tolerances, what kind of parameters are affected by it ?_
The tolerance is the amount (in %) of tolerated difference between the condition image pixels and the screen frame pixels. This option can be useful when trying to detect something:
  * that has a partially transparent background, as things displayed below it can change the color a bit
  * that has an effect on it, such as a glowing effect
  * that can have little effect applied over your detected item, such as particles.


I hope this will helps you. If you have more questions, feel free to ask.

**Comment by Metazolid** [comment]
Awesome, thanks.
I think this also answered my follow-up question on how to save some performance since I've noticed the phone becomes quite hot when using the tool for longer periods of time.
